### ISSUE TITLE
azurerm provider implicitly tries and fails to build an AzureAD client for `azuread_x` data blocks

### ISSUE THREAD

**Issue by ResDiaryLewis** [terraform-providers/terraform-provider-azurerm#10916]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```shell
Terraform v0.14.7
+ provider registry.terraform.io/hashicorp/azuread v1.4.0
+ provider registry.terraform.io/hashicorp/azurerm v2.50.0
```

### Affected Resource(s)

* `azuread_XXXXX` data blocks

### Terraform Configuration Files

```hcl
terraform {
  required_version = "~>0.14"

  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "=2.50.0"
    }

    # UNCOMMENT THIS TO FIX
    # azuread = {
    #   source  = "hashicorp/azuread"
    #   version = "=1.4.0"
    # }
  }
}

provider "azurerm" {
  subscription_id = var.azure_subscription_id
  client_id       = var.azure_client_id
  client_secret   = var.azure_secret
  tenant_id       = var.azure_tenant_id
  features {}
}

# UNCOMMENT THIS TO FIX
# provider "azuread" {
#   client_id       = var.azure_client_id
#   client_secret   = var.azure_secret
#   tenant_id       = var.azure_tenant_id
# }

variable "azure_subscription_id" {
  type    = string
  default = "FILL THIS"
}

variable "azure_client_id" {
  type    = string
  default = "FILL THIS"
}

variable "azure_secret" {
  type    = string
  default = "FILL THIS"
}

variable "azure_tenant_id" {
  type    = string
  default = "FILL THIS"
}

data "azuread_group" "sample" {
  display_name = "sample group"
}
```

### Debug Output

I can provide this if necessary, I'm a bit worried about leaking credentials.

### Expected Behaviour

Either:

1. Terraform should step in and tell the user to initialise the `azuread` provider. This happens for other providers, e.g. if I were to reference an `aws_x` data block. I'm guessing it doesn't because the `azurerm` somehow supports `azuread` data blocks implicitly.
2. The AzureAD client should be built using the same credentials as passed to the `azurerm` provider.

### Actual Behaviour

With the `azuread` provider in the above code commented out, an AzureAD client isn't successfully built. An error is thrown:

```shell
Error: Building AzureAD Client

obtain tenant() from Azure CLI: Error parsing json result from the Azure CLI:
Error waiting for the Azure CLI: exit status 1: Please run 'az login' to setup
account.
```

#### Unhappy Path

1. `terraform init`
2. `az logout` - this is important to ensure that the AzureAD client is not built using your current login
3. Clear any `ARM_x` environment variables, e.g. `ARM_CLIENT_ID`
4. `terraform plan`

#### Happy Path 1

1. `terraform init`
2. `az logout` - this is important to ensure that the AzureAD client is not built using your current login
3. Populate all `ARM_x` environment variables, e.g. `ARM_CLIENT_ID`
4. `terraform plan`
5. The AzureAD client uses the environment variables populated in step 3.

#### Happy Path 2

1. `terraform init`
2. `az login`
3. `terraform plan`
4. The AzureAD client uses the `az` CLI

#### Happy Path 3 

1. Uncomment commented out code
2. `terraform init`
3. `az logout` - this is important to ensure that the AzureAD client is not built using your current login
4. Clear any `ARM_x` environment variables, e.g. `ARM_CLIENT_ID`
5. `terraform plan`
6. The `azuread` provider credentials are correctly used

**Comment by tombuildsstuff** [comment]
hi @ResDiaryLewis 

Thanks for opening this issue.

Resources within the AzureAD Provider (e.g. starting `azuread_`) come from the AzureAD Provider, rather than the AzureRM Provider, as such you'd need to configure the AzureAD Provider when using resources in that provider, making the config:

```hcl
terraform {
  required_version = "~>0.14"

  required_providers {
    azuread = {
      source  = "hashicorp/azuread"
      version = "=1.4.0"
    }
  }
}

provider "azuread" {
  client_id       = var.azure_client_id
  client_secret   = var.azure_secret
  tenant_id       = var.azure_tenant_id
}

variable "azure_subscription_id" {
  type    = string
  default = "FILL THIS"
}

variable "azure_client_id" {
  type    = string
  default = "FILL THIS"
}

variable "azure_secret" {
  type    = string
  default = "FILL THIS"
}

variable "azure_tenant_id" {
  type    = string
  default = "FILL THIS"
}

data "azuread_group" "sample" {
  display_name = "sample group"
}
```

This forum is intended to be used for feature enhancements and bugs in the Azure Provider - so that we can keep this forum focused on that we instead ask that broader questions are raised [using one of the Community Resources](https://www.terraform.io/community.html). As such I'm going to close this issue for the moment, but I believe you should be able to get an answer for this [using one of the Community Resources](https://www.terraform.io/community.html).

Thanks!

**Comment by ResDiaryLewis** [comment]
Hello, @tombuildsstuff - I feel that I covered this argument in the first clause of the _"Expected Behaviour"_ heading:

>Terraform should step in and tell the user to initialise the `azuread` provider. This happens for other providers, e.g. if I were to reference an `aws_x` data block. I'm guessing it doesn't because the `azurerm` somehow supports `azuread` data blocks implicitly.

If these providers were truly separate, then that'd be totally fine as I'd have gotten a helpful message from Terraform itself informing me that I had not initialised the `azuread` provider. Instead I spent the better part of a working day trying to figure out why it wasn't working.

Correct me if I'm wrong, but there must be some sort of cross-over between the two providers if _"Happy Path 1"_ and _"Happy Path Two"_ both work without initialising the `azuread` provider.
